pkg/logs/input/docker/launcher.go

```diff
@@ -55,13 +59,37 @@ func NewLauncher(sources *config.LogSources, services *service.Services, pipelin
 		stop:               make(chan struct{}),
 		erroredContainerID: make(chan string),
 		lock:               &sync.Mutex{},
+		dockerRetrierStop:  make(chan struct{}, 1),
 	}
-	err := launcher.setup()
-	if err != nil {
+
+	var retryStrategy retry.Strategy
```

Comment:
We could probably move all this new logic to setup a retrier to a dedicated method

Reply:
Done. I’ve extracted all the retry logic in a dedicated function in my last commit.